### PR TITLE
Check StatusCode when doing HTTP GET requests

### PR DESCRIPTION
**Pull request by MarcelMue**
New change looks like this in a `t.fatal` when something goes wrong:
```
Got StatusCode 422 with body {"error":{"code":"invalid-release","details":{"release":"stable"},"message":"Invalid requirement specification: u'stable'"}}
		: invalid statuscode
```
Previously no error was thrown and the body was written to a file which was treated as valid output. 

Uncertain if the error could be matched directly but this should at least be an improvement.

### CODE REVIEW

**Review by indyfree — APPROVED** [review]
LGTM. Good to have to HTTP Codes checked.

File: apprclient.go [hunk 1]
```diff
@@ -264,6 +265,14 @@ func (c *Client) doFile(req *http.Request) (string, error) {
 		return "", microerror.Mask(err)
 	}
 	defer resp.Body.Close()
+	if resp.StatusCode != http.StatusOK {
+		buf := new(bytes.Buffer)
+		_, err = buf.ReadFrom(resp.Body)
+		if err != nil {
+			return "", microerror.Mask(err)
+		}
+		return "", microerror.Maskf(invalidStatusCodeError, fmt.Sprintf("Got StatusCode %d with body %s", resp.StatusCode, buf.String()))
```
Comment (kopiczko) [inline apprclient.go]: Nit: it should be lowercase `got`

File: apprclient.go [hunk 2]
```diff
@@ -264,6 +265,14 @@ func (c *Client) doFile(req *http.Request) (string, error) {
 		return "", microerror.Mask(err)
 	}
 	defer resp.Body.Close()
+	if resp.StatusCode != http.StatusOK {
+		buf := new(bytes.Buffer)
+		_, err = buf.ReadFrom(resp.Body)
```
Comment (kopiczko) [inline apprclient.go]: `defer resp.Body.Close()`

File: apprclient.go (hunk 1 above)
Comment (MarcelMue) [reply apprclient.go]: Done.

**Review by kopiczko — APPROVED** [review]

**Pull request merged**